lib/req/steps.ex

```diff
@@ -109,22 +109,22 @@ defmodule Req.Steps do
   @doc """
   Sets base URL for all requests.
 
+  Note this step would overwrite existing base_url, see examples below:
+
   ## Examples
 
       iex> options = [base_url: "https://httpbin.org"]
       iex> Req.get!("/status/200", options).status
       200
       iex> Req.get!("/status/201", options).status
       201
+      iex> Req.get!("https://unknown-host/status/201", options).status
```

Comment:
I don't think we should ignore `https://unknown-host` that the user just given. If anything, I think it should take precedence over `:base_url`. That's what Tesla [1] does too. If we have that behaviour, would that be OK for your use case?

[1] https://github.com/teamon/tesla/blob/v1.4.4/lib/tesla/middleware/core.ex#L35

Reply:
Yes, I agree we shouldn't ignore the URL given by the user

But I'm not sure if silently using that over `:base_url` is a good idea
I'd prefer to raise in this case (which is exactly the old behavior I'm changing here 😂 )

Maybe we should distinguish the `base_url + path` case and the `full_url` case in the `Req` struct?
So that we can raise when `full_url` is set and `put_base_url` gets called, and only override `base_url` when `full_url` is nil

If this sounds too complicated to you, I'm okay with taking `full_url` over `base_url`, as in that case, `retry` would still work with `put_base_url` (#84)

Reply:
> I'm okay with taking full_url over base_url

Yup, let's roll with that!